src/utils/shallowEqual.js

```diff
@@ -1,23 +1,19 @@
-export default function shallowEqual(objA, objB) {
-  if (objA === objB) {
-    return true
-  }
+const hasOwn = Object.prototype.hasOwnProperty
 
-  const keysA = Object.keys(objA)
-  const keysB = Object.keys(objB)
+export default function shallowEqual(a, b) {
+  if (a === b) return true
 
-  if (keysA.length !== keysB.length) {
-    return false
+  let countA = 0
+  let countB = 0
+  
+  for (let key in a) {
+    if (hasOwn.call(a, key) && a[key] !== b[key]) return false
+    countA++
   }
 
-  // Test for A's keys different from B.
-  const hasOwn = Object.prototype.hasOwnProperty
-  for (let i = 0; i < keysA.length; i++) {
-    if (!hasOwn.call(objB, keysA[i]) ||
-        objA[keysA[i]] !== objB[keysA[i]]) {
-      return false
-    }
+  for (let key in b) {
+    if (hasOwn.call(b, key)) countB++
```

Comment:
This could exit before completely iterating through all the `b` keys by checking if `countB` is greater than `countA` in each iteration:

```js
for (let key in b) {
  if (hasOwn.call(b, key) && countA <= countB++) return false
}

return true
```